README.md

```diff
@@ -1,2 +1,217 @@
+while recording images in our simulated environment. 
+```
+https://github.com/IMRCLab/crazyswarm2/tree/blender_viz
+``` 
+
+## 2. Real-world data collection
+### 2.1 GAP8 firmware with Docker
+
+**Installing the bootloader**
+
+One-time for "new" AI decks, then the firmware can be replaced wirelessly.
+
+```
+https://github.com/bitcraze/aideck-gap8-bootloader
+```
+
+**Building and flashing wirelessly**
+
+```
+docker pull bitcraze/aideck
+docker run --rm -v ${PWD}:/module bitcraze/aideck tools/build/make-example examples/other/wifi-img-streamer image
+cfloader flash examples/other/wifi-img-streamer/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/target.board.devices.flash.img deck-bcAI:gap8-fw -w radio://0/80/2M
+```
+
+**Attaching the debugger**
+
+First, connect ARM-JTAG-20-10 ADAPTER to ARM-USB-TINY-H. After, ARM-JTAG-20-10 ADAPTER should get connected to GAP8 JTAG interface on the AI-deck as given [here](https://www.bitcraze.io/documentation/repository/AIdeck_examples/master/getting-started/jtag-programming/). **Note:** Purple color is on the left side when viewing the board from the top and camera front.
+
+**Note** If you have the JTAG adapter attached before powering the board, the camera might not start up properly. Therefore, it might be necessary to unplug the JTAG if the AI-deck/Crazyflie is power cycled and plug it back in after it is powered again.
+
+**Building and Flashing**
+
+**Note:** Before running the following docker commands, make sure that Docker is configured to work without Sudo (non-root) with [these instructions](https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user).
+
+**Note:** WiFi Settings are now configured as part of the STM32 firmware.
+
+
+```
+docker run --rm -v ${PWD}:/module --device /dev/ttyUSB0 --privileged -P bitcraze/aideck tools/build/make-example examples/other/wifi-img-streamer flash
+```
+
+### 2.2 ESP32 firmware with Docker
+
+
+**Attaching the debugger**
+
+Connect ARM-JTAG-20-10 ADAPTER to ARM-USB-TINY-H. Then, ARM-JTAG-20-10 ADAPTER should get connected to NINA JTAG interface on the AI-deck. **Note:** Purple color is on the left side when viewing the board from the top and camera front.
+
+**Initial Set-up**
+```
+docker pull bitcraze/aideck-nina
+```
+**Install IDF**
+```
+git clone https://github.com/espressif/esp-idf.git
+cd esp-idf
+git checkout v4.3.1
+git submodule sync
+git submodule update --init --recursive
+./install.sh
+```
+**Building and Flashing**
+```
+cd aideck-esp-firmware
+source ../esp-idf/export.sh
+idf.py app bootloader
+docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/aideck-nina /bin/bash -c "/openocd-esp32/bin/openocd -f interface/ftdi/olimex-arm-usb-tiny-h.cfg -f board/esp-wroom-32.cfg -c 'program_esp32 build/bootloader/bootloader.bin 0x1000 verify' -c 'program_esp32 build/aideck_esp.bin 0x10000 verify reset exit'"
+```
```

Comment:
can remove this - we just use the latest official esp32 firmware

Reply:
Should mention it here as well ?

Reply:
mention what? that we use official esp32?